crates/holochain_zome_types/src/call.rs

```diff
@@ -5,32 +5,61 @@ use crate::zome::ZomeName;
 use crate::ExternIO;
 use holo_hash::AgentPubKey;
 
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+pub enum CallTargetCell {
+    Other(CellId),
+    Local,
+}
+
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+pub enum CallTarget {
+    Agent(AgentPubKey),
+    Cell(CallTargetCell),
+}
+
 #[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
 pub struct Call {
-    pub to_cell: Option<CellId>,
+    pub target: CallTarget,
     pub zome_name: ZomeName,
     pub fn_name: FunctionName,
     pub cap_secret: Option<CapSecret>,
     pub payload: ExternIO,
-    pub provenance: AgentPubKey,
 }
 
 impl Call {
     pub fn new(
-        to_cell: Option<CellId>,
+        target: CallTarget,
         zome_name: ZomeName,
         fn_name: FunctionName,
         cap_secret: Option<CapSecret>,
         payload: ExternIO,
-        provenance: AgentPubKey,
     ) -> Self {
         Self {
-            to_cell,
+            target,
             zome_name,
             fn_name,
             cap_secret,
             payload,
-            provenance,
         }
     }
+
+    pub fn target(&self) -> &CallTarget {
+        &self.target
+    }
+
+    pub fn zome_name(&self) -> &ZomeName {
+        &self.zome_name
+    }
+
+    pub fn fn_name(&self) -> &FunctionName {
+        &self.fn_name
+    }
+
+    pub fn cap_secret(&self) -> &Option<CapSecret> {
+        &self.cap_secret
+    }
```

Comment:
Usually more useful this way

```suggestion
    pub fn cap_secret(&self) -> Option<&CapSecret> {
        self.cap_secret.as_ref()
    }
```